### PR TITLE
feat: configurable testnet for interlay currencies

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

**Comment by sander2** [comment]
I'm not a fan of using compile-time features for this to be honest

**Comment by gregdhill** [comment]
> I'm not a fan of using compile-time features for this to be honest

Can you offer an alternate suggestion then?

**Comment by sander2** [comment]
Is there no way to include both versions? We probably can't do the customization through the chain spec but maybe there is a way to have 2 runtimes without duplicating _all_ of the code?

**Review by nud3l — APPROVED** [review]

**Comment by nud3l** [comment]
Let's merge for now until we find a better way to do this.

File: parachain/runtime/testnet/src/xcm_config.rs [hunk 1]
```diff
@@ -68,52 +68,49 @@ pub type Barrier = (
...
+pub fn xcm_per_second() -> u128 {
     let base_weight = Balance::from(ExtrinsicBaseWeight::get());
     let base_tx_per_second = (WEIGHT_PER_SECOND as u128) / base_weight;
-    base_tx_per_second * base_tx_in_ksm()
+    base_tx_per_second * base_tx_in_xcm()
 }
 
 parameter_types! {
-    pub KsmPerSecond: (AssetId, u128) = (MultiLocation::parent().into(), ksm_per_second());
+    pub KsmPerSecond: (AssetId, u128) = (MultiLocation::parent().into(), xcm_per_second());
```
Comment (savudani8) [inline parachain/runtime/testnet/src/xcm_config.rs:86]: This should be `pub XcmPerSecond` then, right?

**Pull request merged**